src/main/app/form/Form.ts

```diff
@@ -14,7 +14,7 @@ export class Form {
       .filter((key) => this.form.fields[key].validator !== undefined)
       .reduce((errors: FormError[], propertyName: string) => {
         const field = this.form.fields[propertyName];
-        const errorType = field.validator!(body[propertyName]);
+        const errorType = field.validator?.(body[propertyName] as string);
```

Comment:
so is the ? the same as ! operator?

Reply:
Not quite:

* `!.` [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) (TypeScript only). It's a way to tell the compiler that "this expression cannot be `null` or `undefined` here" (so don't show any errors)
* `?.` [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (works in both TS & JS). Returns `undefined` if the property or method doesn't exist

Reply:
thanks for the explainations, in that case, shouldnt ! still be used as we can filtering that field and checking its not undefined before hand. I guess its fine either way, so im good either way.

Reply:
IMO I think ! is fine because of the filter but I'm happy either way

Reply:
Our current eslint rules from the TypeScript recommended ruleset, mark this line as a `Forbidden non-null assertion. eslint(@typescript-eslint/no-non-null-assertion)`.

Possible because if you're saying that the `field.validator` field might not exist, but then trying to use it by force. If it is actually optional then you'll get an error trying to invoke a non-existent function. Using `?.` would prevent this error.

However, maybe a better option would be to mark this field as required. I'll take a look if that could work.

Reply:
no I'm happy with ?